warnings when compiling elektra

I just recompiled elektra for OpenWRT (32bit arm target) and saved a log of the warnings. Maybe somebody wants to look at them and clean things up a bit:
http://friends.ccbib.org/lambda/elektra.warn

```
  /home/lambda/olinuxino/openwrt/build_dir/target-arm_arm926ej-s_musl-1.1.11_eabi/elektra-0.8.17/src/plugins/resolver/filename.c:79:43: warning: format '%d' expects argument of type 'int', but argument 6 has ty    pe 'suseconds_t {aka long int}' [-Wformat=]
    snprintf (where + len, POSTFIX_SIZE - 1, ".%d:%ld." ELEKTRA_TIME_USEC_F ".tmp", getpid (), tv.tv_sec, tv.tv_usec);
                                             ^
```
seems to be a false positive / preprocessor problem

not sure what todo about this one:
```
/home/lambda/olinuxino/openwrt/staging_dir/target-arm_arm926ej-s_musl-1.1.11_eabi/usr/include/boost/type_traits/detail/bool_trait_def.hpp:18:79: note: #pragma message: NOTE: Use of this header (bool_trait_def    .hpp) is deprecated
 # pragma message("NOTE: Use of this header (bool_trait_def.hpp) is deprecated")
```

--- a/src/libs/meta/meta.c
+++ b/src/libs/meta/meta.c
@@ -1383,7 +1383,7 @@ TopSortCleanup:
 	ksDel (done);
 	keyDel (orderCounter);
 	elektraFree (localArray);
-	for (unsigned int j = 0; j < size; ++j)
+	for (ssize_t j = 0; j < size; ++j)
 	{
 		elektraFree (adjMatrix[j].deps);
 	}

--- a/src/plugins/filecheck/filecheck.c
+++ b/src/plugins/filecheck/filecheck.c
@@ -247,7 +247,7 @@ static long checkFile (Key * parentKey, const char * filename, checkStruct * che
 			le_ret = validateLineEnding (line, &(checkConf->validLE), 0);
 			if (le_ret)
 			{
-				ELEKTRA_SET_ERRORF (137, parentKey, "invalid lineending at position %lu", bytesRead + le_ret);
+				ELEKTRA_SET_ERRORF (137, parentKey, "invalid lineending at position %zd", bytesRead + le_ret);
 				retVal = -1;
 				break;
 			}
@@ -257,7 +257,7 @@ static long checkFile (Key * parentKey, const char * filename, checkStruct * che
 			null_ret = checkNull (line, bytesRead);
 			if (null_ret)
 			{
-				ELEKTRA_SET_ERRORF (137, parentKey, "found null-byte at position %lu", bytesRead + null_ret);
+				ELEKTRA_SET_ERRORF (137, parentKey, "found null-byte at position %zd", bytesRead + null_ret);
 				retVal = -1;
 				break;
 			}
@@ -267,7 +267,7 @@ static long checkFile (Key * parentKey, const char * filename, checkStruct * che
 			iconv_ret = validateEncoding (line, conv, bytesRead);
 			if (iconv_ret)
 			{
-				ELEKTRA_SET_ERRORF (137, parentKey, "invalid encoding at position %lu", bytesRead + iconv_ret);
+				ELEKTRA_SET_ERRORF (137, parentKey, "invalid encoding at position %zd", bytesRead + iconv_ret);
 				retVal = -1;
 				break;
 			}
@@ -288,7 +288,7 @@ static long checkFile (Key * parentKey, const char * filename, checkStruct * che
 			unprintable_ret = checkUnprintable (line);
 			if (unprintable_ret)
 			{
-				ELEKTRA_SET_ERRORF (137, parentKey, "unprintable character at position %lu", bytesRead + unprintable_ret);
+				ELEKTRA_SET_ERRORF (137, parentKey, "unprintable character at position %zd", bytesRead + unprintable_ret);
 				retVal = -1;
 				break;
 			}

